### DISCUSSION TITLE
`parseDuration()` seconds constraint

### DISCUSSION THREAD

**Discussion by crazko** [adobe/react-spectrum#6371] [General]
`parseDuration()` of `@internationalized/date` sets boundaries to the numbers that are being parsed.

[Seconds are set to max 59](https://github.com/adobe/react-spectrum/blob/cf0846eb49fcdde669e9dc4c0c2d2109b50e46dd/packages/%40internationalized/date/src/string.ts#L256) which prevents from parsing duration with fractional seconds above this limit:

```js
const duration = "PT59.123789456S";
const result = parseDuration(duration); // throws: Invalid ISO 8601 Duration string: PT59.123789456S
```

While testing this with Temporal (polyfill) results in milli/micro/nano seconds:

```js
const result = Temporal.Duration.from(duration);

// result.seconds - 59
// result.milliseconds - 123
// result.microseconds - 789
// result.nanoseconds - 456
```

Is this a desired behavior? Shouldn't it be allowed to have any decimal seconds value up to the 60?

Link to [CodeSandBox](https://codesandbox.io/p/devbox/nervous-williamson-ml3ttl?file=%2Findex.js).

---

Related: https://github.com/adobe/react-spectrum/issues/5416

**Comment by ktabors** [comment]
Thanks for the feedback and the CodeSandbox reproduction. `parseDuration()` from `@internationalized/date` should match the behavior of Temporal.Duration. This would be a good first issue for someone to contribute as a PR.
